core/core_isolate.rs

```diff
@@ -71,8 +73,27 @@ pub enum StartupData<'a> {
   None,
 }
 
+impl StartupData<'_> {
+  fn into_options(self) -> (Option<OwnedScript>, Option<Snapshot>) {
+    match self {
+      Self::Script(script) => (Some(script.into()), None),
+      Self::Snapshot(snapshot) => (None, Some(snapshot)),
+      Self::None => (None, None),
+    }
+  }
+}
+
 type JSErrorCreateFn = dyn Fn(JSError) -> ErrBox;
 
+type PinnedRefCell = Pin<Box<RefCell<dyn Any>>>;
```

Comment:
Curious question - what does a Pin around a RefCell (boxed or not) achieve?
I'd think that, since a RefCell allows interior mutability, this kinda negates the effect of the pin.

Reply:
good point. removed.